File: Borealis.Web/src/app/dtos/station/stationUpdateDto.ts
import { NeighborhoodDto } from "../density/neighborhoodDto";

export class StationUpdateDto{
    public id:number;
    public name:string;
    public address:string;
    public startDate:Date;
    public endDate:Date;
    public latitude:string;
    public longitude:string;
    public altitude:number;
    public neighborhood:NeighborhoodDto;
    public neighborhoodId= ()=> this.neighborhood.id;

    constructor(id:number, name:string, address:string, startDate:Date, endDate:Date, latitude:string, longitude:string, altitude:number) {
        this.id = id;
        this.name = name;
        this.address = address;
        this.startDate = startDate;
        this.endDate = endDate;
        this.latitude = latitude;
        this.longitude = longitude;
        this.altitude = altitude;
    }
}

File: Borealis.Api/app/dto/station/station_update_dto.py
class StationUpdateDto():
    def __init__(self, name, address, start_date, end_date, latitude, longitude, altitude, neighborhood_id):
        self.name = name
        self.address = address
        self.start_date = start_date
        self.end_date = end_date
        self.latitude = latitude
        self.longitude = longitude
        self.altitude = altitude
        self.neighborhood_id = neighborhood_id

